src/dialogs/config-flow/step-flow-form.ts

```diff
@@ -161,6 +151,30 @@ class StepFlowForm extends LitElement {
     }
   }
 
+  /**
+   * If a field is undefined it was emptied by the user.
+   *
+   * If their schema allows none set those to null so they won't be overridden
+   * by voluptuous defaults in core.
+   * @param data The config flow payload
+   * @returns the same data where undefined fields are null if they allow it
+   */
+  private _transformEmptiedFields<T>(
+    data: Record<string, T>
+  ): Record<string, T | null> {
+    const toSendData: Record<string, T | null> = data;
+
+    this.step.data_schema
+      .filter((field) => field.allow_none === true)
+      .forEach((field) => {
+        if (toSendData[field.name] === undefined) {
```

Comment:
We no longer handle empty strings now, should we add those too?

Reply:
If the field is an empty string it was sent by the backend as the default value and not changed by the user. As soon as the user enters characters and removes them again it becomes `undefined` if the user does not touch the field it is an empty string.

At least thats the behaviour I saw when testing the feature.